### PR TITLE
Fixed issue in getting license agreement

### PR DESCRIPTION
**Pull request by hasuniea**

### CODE REVIEW

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java [hunk 1]
```diff
@@ -438,58 +439,61 @@ public void generateComplianceFeatureStatus(SyncmlDocument syncmlDocument) throw
...
+                        getProfileFeaturesList() != null) {
+                    List<ProfileFeature> profileFeatures = WindowsAPIUtils.getPolicyManagerService().
+                            getAppliedPolicyToDevice(deviceIdentifier).getProfile().getProfileFeaturesList();
+                    List<ComplianceFeature> complianceFeatures = new ArrayList<>();
+                    for (ProfileFeature activeFeature : profileFeatures) {
+                        JSONObject policyContent = new JSONObject(activeFeature.getContent().toString());
+
+                        for (Profile deviceFeature : profiles) {
+                            if (deviceFeature.getFeatureCode().equals(activeFeature.getFeatureCode()) &&
+                                    deviceFeature.getFeatureCode().equals(PluginConstants.OperationCodes.CAMERA)) {
```
Comment (harshanL) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java]: Fix equals method.

Comment (hasuniea) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java]: fixed the issue

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java [hunk 2]
```diff
@@ -438,58 +439,61 @@ public void generateComplianceFeatureStatus(SyncmlDocument syncmlDocument) throw
...
-                                deviceFeature.getFeatureCode().equals(PluginConstants.OperationCodes.
-                                        ENCRYPT_STORAGE)) {
-                            if (policyContent.getBoolean(PluginConstants.PolicyConfigProperties.
-                                    ENCRYPTED_ENABLE) == (deviceFeature.isEnable())) {
-                                isCompliance = true;
-                                deviceFeature.setCompliance(isCompliance);
-                            } else {
-                                deviceFeature.setCompliance(isCompliance);
+                            if (deviceFeature.getFeatureCode().equals(activeFeature.getFeatureCode()) &&
+                                    deviceFeature.getFeatureCode().equals(PluginConstants.OperationCodes.
```
Comment (prabathabey) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java]: As a best practice, always compare non-deterministic variable values against known constants.

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java [hunk 3]
```diff
@@ -438,58 +439,61 @@ public void generateComplianceFeatureStatus(SyncmlDocument syncmlDocument) throw
...
-                                deviceFeature.getFeatureCode().equals(PluginConstants.OperationCodes.
-                                        PASSCODE_POLICY)) {
-                            if (policyContent.getBoolean(PluginConstants.PolicyConfigProperties.
-                                    ENABLE_PASSWORD) == (deviceFeature.isEnable())) {
-                                isCompliance = true;
-                                deviceFeature.setCompliance(isCompliance);
-                            } else {
-                                deviceFeature.setCompliance(isCompliance);
+                            if (deviceFeature.getFeatureCode().equals(activeFeature.getFeatureCode()) &&
+                                    deviceFeature.getFeatureCode().equals(PluginConstants.OperationCodes.
```
Comment (prabathabey) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java]: As a best practice, always compare non-deterministic variable values against known constants.

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/configurationmgtservice/ConfigurationMgtService.java [hunk 4]
```diff
@@ -108,25 +109,30 @@ public Message ConfigureSettings(TenantConfiguration configuration) throws Windo
...
+        TenantConfiguration tenantConfiguration;
+        List<ConfigurationEntry> configs;
         try {
-            if (WindowsAPIUtils.getDeviceManagementService().
-                    getConfiguration(DeviceManagementConstants.MobileDeviceTypes.MOBILE_DEVICE_TYPE_WINDOWS) != null) {
-                tenantConfiguration = WindowsAPIUtils.getDeviceManagementService().
+            tenantConfiguration = WindowsAPIUtils.getDeviceManagementService().
                     getConfiguration(DeviceManagementConstants.MobileDeviceTypes.MOBILE_DEVICE_TYPE_WINDOWS);
-            List<ConfigurationEntry> configs = tenantConfiguration.getConfiguration();
+            if(tenantConfiguration != null) {
```
Comment (prabathabey) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/configurationmgtservice/ConfigurationMgtService.java]: Fix formatting issues.

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/configurationmgtservice/ConfigurationMgtService.java [hunk 5]
```diff
@@ -108,25 +109,30 @@ public Message ConfigureSettings(TenantConfiguration configuration) throws Windo
...
+
             ConfigurationEntry entry = new ConfigurationEntry();
             License license = WindowsAPIUtils.getDeviceManagementService().getLicense(
-                    DeviceManagementConstants.MobileDeviceTypes.MOBILE_DEVICE_TYPE_WINDOWS, PluginConstants.
-                            TenantConfigProperties.LANGUAGE_US);
-            if(license != null) {
+                    DeviceManagementConstants.MobileDeviceTypes.MOBILE_DEVICE_TYPE_WINDOWS,
+                    PluginConstants.TenantConfigProperties.LANGUAGE_US);
+
+            if(license != null && configs != null) {
```
Comment (prabathabey) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/configurationmgtservice/ConfigurationMgtService.java]: Fix formatting issues.

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java (hunk 2 above)
Comment (hasuniea) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java]: already fixed it from https://github.com/wso2/product-mdm/pull/151

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java (hunk 3 above)
Comment (hasuniea) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java]: already fixed it from  https://github.com/wso2/product-mdm/pull/151

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/configurationmgtservice/ConfigurationMgtService.java (hunk 4 above)
Comment (hasuniea) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/configurationmgtservice/ConfigurationMgtService.java]: fixed from https://github.com/wso2/product-mdm/pull/156

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/configurationmgtservice/ConfigurationMgtService.java (hunk 5 above)
Comment (hasuniea) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/configurationmgtservice/ConfigurationMgtService.java]: fixed from https://github.com/wso2/product-mdm/pull/156

**Pull request merged**